packages/headless/src/features/configuration/configuration-slice.ts

```diff
@@ -17,13 +17,28 @@ import {
   analyticsAPIEndpoint,
 } from './configuration-state';
 
-function analyticsUrlFromPlatformUrl(platformUrl: string) {
-  const isCoveoURL = platformUrl.match(
-    /^https:\/\/platform(dev|qa|hipaa)?(-)?(eu|au)?\.cloud\.coveo\.com/
-  );
-  if (isCoveoURL) {
-    return platformUrl.replace(/^(https?:\/\/)platform/, '$1analytics');
+function analyticsUrlFromPlatformUrl(
+  platformUrl: string,
+  organizationId: string
+) {
+  const isCoveoPlatformURL =
+    /^https:\/\/platform(dev|qa|hipaa)?(-)?(eu|au)?\.cloud\.coveo\.com/.test(
+      platformUrl
+    );
+  if (isCoveoPlatformURL) {
+    return platformUrl.replace(/^(https:\/\/)platform/, '$1analytics');
   }
+
+  const isCoveoOrgDomainUrl = new RegExp(
```

Comment:
You should be able to achieve the same thing by building only one regex (the second one, with the capturing group).

Suggestion: 
```
const isCoveoOrgDomainUrlMatch = platformUrl.match(new RegExp(`^https://(${organizationId}\\.org)\\.coveo.com`))

if(isCoveoOrgDomainUrlMatch) {
   return platformUrl.replace(isCoveoOrgDomainUrlMatch[1], 'analytics.cloud')
}